File: frontend/lib/models/delivery_zone.dart
class DeliveryZone {
  final String id;
  final String name;
  final String? countries;
  final double fee;
  final double? minOrderAmount;
  final int? estimatedDays;
  final bool isActive;

  DeliveryZone({
    required this.id,
    required this.name,
    this.countries,
    this.fee = 0,
    this.minOrderAmount,
    this.estimatedDays,
    this.isActive = true,
  });

  factory DeliveryZone.fromJson(Map<String, dynamic> json) => DeliveryZone(
    id: json['id'].toString(),
    name: json['name'] ?? '',
    countries: json['countries'],
    fee: (json['fee'] ?? 0).toDouble(),
    minOrderAmount: (json['minOrderAmount'] as num?)?.toDouble(),
    estimatedDays: json['estimatedDays'],
    isActive: json['isActive'] ?? true,
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    if (countries != null) 'countries': countries,
    'fee': fee,
    if (minOrderAmount != null) 'minOrderAmount': minOrderAmount,
    if (estimatedDays != null) 'estimatedDays': estimatedDays,
    'isActive': isActive,
  };
}

File: backend/src/main/java/io/makewebsite/entity/DeliveryZone.java
package io.makewebsite.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "delivery_zones")
public class DeliveryZone {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "boutique_id")
    private Boutique boutique;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String countries;

    @Builder.Default
    private Double fee = 0.00;

    @Column(name = "min_order_amount")
    private Double minOrderAmount;

    @Builder.Default
    @Column(name = "estimated_days")
    private Integer estimatedDays = 3;

    @Builder.Default
    @Column(name = "is_active")
    private Boolean isActive = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
